File: apps/mobile/ios/ParentAssist/Services/SessionService.swift
import Foundation

protocol SessionServicing {
    func createSession() async throws -> ShareSession
    func stopSession(_ session: ShareSession) async
}

struct MockSessionService: SessionServicing {
    private let clock: () -> Date

    init(clock: @escaping () -> Date = Date.init) {
        self.clock = clock
    }

    func createSession() async throws -> ShareSession {
        let sessionId = UUID().uuidString
        let joinCode = String(sessionId.prefix(6)).uppercased()
        let expiresAt = clock().addingTimeInterval(60 * 15)
        let url = URL(string: "https://example.com/join/\(joinCode)")!
        return ShareSession(sessionId: sessionId, joinCode: joinCode, expiresAt: expiresAt, viewerUrl: url)
    }

    func stopSession(_ session: ShareSession) async {
        _ = session
    }
}

File: apps/mobile/android/app/src/main/java/com/example/parentassist/SessionService.kt
package com.example.parentassist

import java.time.Clock
import java.time.Duration
import java.time.Instant
import java.util.UUID

interface SessionService {
    suspend fun createSession(): ShareSession
    suspend fun stopSession(session: ShareSession)
}

class MockSessionService(
    private val clock: Clock = Clock.systemUTC()
) : SessionService {
    override suspend fun createSession(): ShareSession {
        val sessionId = UUID.randomUUID().toString()
        val joinCode = sessionId.take(6).uppercase()
        val expiresAt = Instant.now(clock).plus(Duration.ofMinutes(15))
        val viewerUrl = "https://example.com/join/$joinCode"
        return ShareSession(sessionId, joinCode, expiresAt, viewerUrl)
    }

    override suspend fun stopSession(session: ShareSession) {
        session.toString()
    }
}
